pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/channel/MailboxContentObserver.java

```diff
@@ -47,11 +48,13 @@ public class MailboxContentObserver implements StreamObserver<MailboxContent> {
   private final MailboxService _mailboxService;
   private final StreamObserver<MailboxStatus> _responseObserver;
 
+  private final List<ByteBuffer> _mailboxBuffers;
   private transient ReceivingMailbox _mailbox;
 
   public MailboxContentObserver(MailboxService mailboxService, StreamObserver<MailboxStatus> responseObserver) {
     _mailboxService = mailboxService;
     _responseObserver = responseObserver;
+    _mailboxBuffers = new ArrayList<>();
```

Comment:
(MAJOR) Is this thread safe? Are we sharing the same observer across multiple queries?

Reply:
As we discussed offline, Mailbox Observer is not shared among concurrent queries and therefore is safe to just stack blocks until a block with `waitForMore=false` is received.

Anyway, we will add an integration test with a small buffer and concurrent queries to verify that.